nixos/modules/services/networking/zerotierone.nix

```diff
@@ -22,12 +32,13 @@ in
       path = [ cfg.package ];
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
-      preStart =
-        ''
+      preStart = ''
         mkdir -p /var/lib/zerotier-one
         chmod 700 /var/lib/zerotier-one
         chown -R root:root /var/lib/zerotier-one
-        '';
+      '' + optionalString (cfg.joinNetwork != null) ''
+        touch "/var/lib/zerotier-one/networks.d/${cfg.joinNetwork}.conf"
```

Comment:
Also, the `networks.d` subdirectory would not exist on the first startup of zerotier. Maybe just add it to the `mkdir -p` above?

Reply:
Yup just noticed that too. Done.